cpu/esp32/irq_arch.c

```diff
@@ -120,3 +173,80 @@ esp_err_t esp_intr_free(intr_handle_t handle)
 {
     return esp_intr_disable(handle);
 }
+
+int esp_intr_get_cpu(intr_handle_t handle)
+{
+    return PRO_CPU_NUM;
+}
+
+#if __riscv
```

Comment:
Unfortunatly not. I tried but since RISC-V ESP SoC add an interrupt controller, it has to be done mor than to set/reset the `MIE` bit in register `mstatus` the RISC-V core.

Reply:
There is no standard RISC-V interrupt controller anyway, currently `riscv_common` has support for PLIC and CLIC, I'm sure another implementation could be added.

Reply:
Hm, getting/set the value of `INTERRUPT_CORE0_CPU_INT_THRESH_REG` in `irq_disable`/`irq_restore` is a very special implementation, apparently not so common that it would be worth implementing in `cpu/riscv_common`. I have to admit that I'm not sure if it is really necessary to preserve the current interrupt priority level, but the functions and libraries of ESP-IDF use different interrupt priorities.

I think another fact also avoids using `cpu/risc_common`, even though it is not yet part of this PR. Including `irq_arch.h` from `cpu/riscv_common` would require adding the include path for the header, but the `thread_arch.h` from `cpu/riscv_common` definitely cannot be used because context frame structure used in ESP-IDF is different and the context switch is realized in a different way.